proxy/src/main/java/com/google/common/util/concurrent/RecyclableRateLimiter.java

```diff
@@ -0,0 +1,138 @@
+package com.google.common.util.concurrent;
+
+import com.google.common.math.LongMath;
+
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * An alternative RateLimiter implementation that allows to "return" unused permits back to the pool to handle retries
```

Comment:
Unfortunately, standard guava rate limiter has a couple of limitations:
- it doesn't allow accumulating "credits" over a time period other than 1 second - the SmoothBursty constructor where you can specify maxBurstSeconds is package-private, and   RateLimiter.create invokes the the constructor with a hardcoded value of "1.0" for maxBurstSeconds. The only way around it is to have a complete implementation.
- it doesn't allow querying the status and seeing if / how many permits are currently available. Having this functionality makes it possible to have only one rate limit. The alternative would be to have separate rate limits for incoming points and for the retry queue, but users don't care (and shouldn't care) about the difference, as they just want to control the rate of points leaving the proxy. With separate rate limiters in place slightly different circumstances can produce vastly different results, so the goal was to have a single limiter.